indico/modules/events/timetable/util.py

```diff
@@ -196,3 +217,52 @@ def render_session_timetable(session, timetable_layout=None, management=False):
     event_info = serialize_event_info(session.event_new)
     tpl = get_template_module('events/timetable/_timetable.html')
     return tpl.render_timetable(timetable_data, event_info, timetable_layout=timetable_layout, management=management)
+
+
+def get_top_timetable_session_blocks(event, day):
+    """Returns a list of event top-level session blocks for the
+       given `day`
+    """
+    return (event.timetable_entries
+            .filter(db.cast(TimetableEntry.start_dt.astimezone(event.tzinfo), db.Date) == day.date())
+            .filter(TimetableEntry.type == TimetableEntryType.SESSION_BLOCK, TimetableEntry.parent_id.is_(None))
+            .all())
+
+
+def move_timetable_entry(entry, parent_entry=None, move_to_day=None):
+    """Moves the `entry` to another session or top-level timetable
+
+    :param entry: `TimetableEntry` to be moved
+    :param parent_entry: If specified then the entry will be set as a child
+                         of parent_entry
+    :param move_to_day: If specified then the entry will be moved to the
+                        top-level timetable on this day
+    """
+    if bool(parent_entry) + bool(move_to_day) != 1:
+        raise Exception("Wrong number of arguments")
+
+    updates = {}
+    contrib_update_data = {}
+    if move_to_day:
+        old_start_time = entry.start_dt.time()
+        new_start_dt = entry.start_dt.replace(hour=old_start_time.hour, minute=old_start_time.minute)
+        updates['start_dt'] = new_start_dt.astimezone(entry.event_new.tzinfo)
+        updates['parent'] = None
+        contrib_update_data = {'session_id': None, 'session_block_id': None}
+    elif parent_entry:
+        children = sorted(parent_entry.children, key=attrgetter('end_dt'), reverse=True)
+        if children:
+            new_start_dt = children[0].end_dt
+        else:
+            new_start_dt = parent_entry.start_dt
+        updates['parent'] = parent_entry
+        updates['start_dt'] = new_start_dt
+        contrib_update_data = {'session_id': parent_entry.session_block.session.id,
+                               'session_block_id': parent_entry.session_block.id}
+
+    if entry.type == TimetableEntryType.CONTRIBUTION:
+        update_contribution(entry.object, contrib_update_data)
+    update_timetable_entry(entry, updates)
```

Comment:
Can this util be used to move a session block? If yes you need to add a new argument to `update_timetable_entry` whether to move child entries too and if it's true (you would set it to true here) you use `entry.move(new_start_dt)` instead of just assigning the new start dt